bindings/go/src/fdb/tuple/tuple.go

```diff
@@ -313,12 +370,95 @@ func (p *packer) encodeTuple(t Tuple, nested bool) {
 // Tuple satisfies the fdb.KeyConvertible interface, so it is not necessary to
 // call Pack when using a Tuple with a FoundationDB API function that requires a
 // key.
+//
+// This method will panic if it contains an incomplete Versionstamp. Use
+// PackWithVersionstamp instead.
+//
 func (t Tuple) Pack() []byte {
-	p := packer{buf: make([]byte, 0, 64)}
+	p := newPacker()
 	p.encodeTuple(t, false)
 	return p.buf
 }
 
+// PackWithVersionstamp packs the specified tuple into a key for versionstamp
+// operations. See Pack for more information. This function will return an error
+// if you attempt to pack a tuple with more than one versionstamp. This function will
+// return an error if you attempt to pack a tuple with a versionstamp position larger
+// than an uint16 on apiVersion < 520.
+func (t Tuple) PackWithVersionstamp(prefix []byte) ([]byte, error) {
+	hasVersionstamp, err := t.HasIncompleteVersionstamp()
+	if err != nil {
+		return nil, err
+	}
+
+	if hasVersionstamp == false {
+		return nil, errors.New("No incomplete versionstamp included in tuple pack with versionstamp")
+	}
+
+	p := newPacker()
+
+	prefixLength := int32(0)
+	if prefix != nil {
+		prefixLength = int32(len(prefix))
+		p.putBytes(prefix)
+	}
+
+	p.encodeTuple(t, false)
+
+	if hasVersionstamp {
+		var scratch [4]byte
+		var offsetIndex int
+
+		apiVersion := fdb.MustGetAPIVersion()
+		if apiVersion < 520 {
+			if p.versionstampPos > math.MaxUint16 {
+				return nil, errors.New("Versionstamp position too large")
+			}
+
+			offsetIndex = 1
+			binary.LittleEndian.PutUint16(scratch[:], uint16(prefixLength+p.versionstampPos))
+		} else {
+			offsetIndex = 3
+			binary.LittleEndian.PutUint32(scratch[:], uint32(prefixLength+p.versionstampPos))
+		}
+
+		p.putBytes(scratch[0:offsetIndex])
+	}
+
+	return p.buf, nil
+}
+
+// HasIncompleteVersionstamp determines if there is at least one incomplete
+// versionstamp in a tuple. This function will return an error this tuple has
+// more than one versionstamp.
+func (t Tuple) HasIncompleteVersionstamp() (bool, error) {
+	incompleteCount := t.countIncompleteVersionstamps()
+
+	var err error
+	if incompleteCount > 1 {
+		err = errors.New("Tuple can only contain one incomplete versionstamp")
+	}
+
+	return incompleteCount == 1, err
```

Comment:
```suggestion
	return incompleteCount >= 1, err
```

Maybe it doesn't matter, but this would mean that if the user looks at a tuple with two versionstamps, then the boolean they get back is always correct regardless of whether there was an error returned as well.